src/stan_math_backend/Stan_math_code_gen.ml

```diff
@@ -711,37 +730,25 @@ using namespace stan::math; |}
 *)
 let custom_functions =
   {|
-template <typename T, typename S>
-std::vector<T> resize_to_match__(std::vector<T>& dst, const std::vector<S>& src) {
+
+template <typename T, typename S, stan::require_all_vector_like_t<T, S>* = nullptr>
+inline decltype(auto) resize_to_match__(T&& dst, S&& src) {
   dst.resize(src.size());
-  return dst;
+  return std::forward<T>(dst);
 }
 
 template <typename T>
-Eigen::Matrix<T, -1, -1>
+inline Eigen::Matrix<T, -1, -1>&
 resize_to_match__(Eigen::Matrix<T, -1, -1>& dst, const Eigen::Matrix<T, -1, -1>& src) {
   dst.resize(src.rows(), src.cols());
   return dst;
 }
 
-template <typename T>
-Eigen::Matrix<T, 1, -1>
-resize_to_match__(Eigen::Matrix<T, 1, -1>& dst, const Eigen::Matrix<T, 1, -1>& src) {
-  dst.resize(src.size());
-  return dst;
-}
-
-template <typename T>
-Eigen::Matrix<T, -1, 1>
-resize_to_match__(Eigen::Matrix<T, -1, 1>& dst, const Eigen::Matrix<T, -1, 1>& src) {
-  dst.resize(src.size());
-  return dst;
-}
-std::vector<double> to_doubles__(std::initializer_list<double> x) {
+inline std::vector<double> to_doubles__(std::initializer_list<double> x) {
   return x;
 }
 
-std::vector<stan::math::var> to_vars__(std::initializer_list<stan::math::var> x) {
+inline std::vector<stan::math::var> to_vars__(std::initializer_list<stan::math::var> x) {
```

Comment:
Would you mind just removing `to_vars__`, `to_doubles__` and all `resize_to_match__`. Sorry for asking you to remove this after doing all this work on them. 
I triple checked and could not find these used in statement gen, stan math code-gen anywhere.

validate_positive_index, validate_unit_vector_index are used but should be in math/stan. But that is a separate issue/PR.

Reply:
Yes good call! I didn't add them originally so if we don't need them I'm fine with getting rid of them